cdc/sink/mysql/causality.go

```diff
@@ -53,21 +60,34 @@ func (c *causality) reset() {
 // detectConflict detects whether there is a conflict
 func (c *causality) detectConflict(keys [][]byte) (bool, int) {
 	if len(keys) == 0 {
-		return false, 0
+		return false, noConflicting
 	}
 
-	firstIdx := -1
+	conflictingWorkerIndex := noConflicting
 	for _, key := range keys {
-		if idx, ok := c.relations[string(key)]; ok {
-			if firstIdx == -1 {
-				firstIdx = idx
-			} else if firstIdx != idx {
-				return true, -1
+		if workerIndex, ok := c.relations[string(key)]; ok {
+			// The first conflict occurred.
+			if conflictingWorkerIndex == noConflicting {
+				conflictingWorkerIndex = workerIndex
+			} else
+			// A second conflict occurs, and it is with another worker.
+			// For example:
+			// txn0[a,b,c] --> worker0
+			// txn1[t,f] --> worker1
+			// txn2[a,f] --> ?
+			// In this case, if we distribute the transaction,
+			// there is no guarantee that it will be executed
+			// in the order it was sent to the worker,
+			// so we have to wait for the previous transaction to finish writing.
+			if conflictingWorkerIndex != workerIndex {
+				return true, multipleConflicting
 			}
 		}
 	}
 
-	return firstIdx != -1, firstIdx
+	// 1) no conflict
+	// 2) conflict with the same worker
+	return conflictingWorkerIndex != noConflicting, conflictingWorkerIndex
```

Comment:
Maybe better to separate conflictingWorkerIndex to conflictFlag and workerIndex?

Reply:
I think it's really just coupled together, and the fact that we distinguish it internally doesn't change the semantics of that function. So I don't think it's too much of a problem to just leave it that way. Unless we're aiming to refactor it. Then we should really redesign the API.